GenderPayGap.WebUI/Controllers/Admin/AdminDatabaseIntegrityChecksController.cs

```diff
@@ -0,0 +1,201 @@
+        public async Task<IActionResult> OrganisationsWithMultipleActiveScopesForASingleYearAjax()
+        {
+            var model = new DatabaseIntegrityChecksViewModel();
+            List<Organisation> organisations = dataRepository.GetAll<Organisation>()
+                .Include(o => o.OrganisationScopes)
+                .ToList();
+            foreach (Organisation organisation in organisations)
+            {
+                List<OrganisationScope> activeScopes = organisation.OrganisationScopes
+                    .Where(os => os.Status == ScopeRowStatuses.Active)
+                    .ToList();
+
+                bool multipleActiveScopesForYear = activeScopes
+                    .GroupBy(scope => scope.SnapshotDate)
+                    .Any(g => g.Count() > 1);
+
+                if (multipleActiveScopesForYear)
+                {
+                    model.OrganisationsWithMultipleActiveScopesForASingleYear.Add(organisation);
+                }
+            }
+
+            return PartialView("OrganisationsWithMultipleActiveScopesForASingleYear", model);
+        }
+
+        [HttpGet("organisations-with-no-active-scope-for-every-year-ajax")]
+        public async Task<IActionResult> OrganisationsWithNoActiveScopeForEveryYearAjax()
+        {
+            var model = new DatabaseIntegrityChecksViewModel();
+            List<Organisation> organisations = dataRepository.GetAll<Organisation>()
+                .Include(o => o.OrganisationScopes)
+                .ToList();
+
+            foreach (Organisation organisation in organisations)
+            {
+                DateTime organisationCreationDate = organisation.Created;
+                DateTime accountingDateForCreationYear = organisation.SectorType.GetAccountingStartDate(organisationCreationDate.Year);
+
+                int firstRequiredAccountingDateYear = organisationCreationDate < accountingDateForCreationYear
+                    ? accountingDateForCreationYear.AddYears(-1).Year
+                    : accountingDateForCreationYear.Year;
+
+                int latestRequiredAccountingDateYear = organisation.SectorType.GetAccountingStartDate().Year;
+
+                List<int> requiredYears = Enumerable.Range(
+                        firstRequiredAccountingDateYear,
+                        (latestRequiredAccountingDateYear - firstRequiredAccountingDateYear) + 1)
+                    .ToList();
+
+                List<int> yearsWithActiveScopes = organisation.OrganisationScopes
+                    .Where(os => os.Status == ScopeRowStatuses.Active)
+                    .GroupBy(scope => scope.SnapshotDate)
+                    .Select(g => g.Key.Year)
+                    .ToList();
+
+                foreach (int year in requiredYears)
+                {
+                    if (!yearsWithActiveScopes.Contains(year))
+                    {
+                        model.OrganisationsWithNoActiveScopeForEveryYear.Add(organisation);
```

Comment:
This might add an organisation multiple times, if several years are missing

Reply:
I think this should solve it ```if (!yearsWithActiveScopes.Contains(year) && !organisationsWithNoActiveScopeForEveryYear.Contains(organisation))
                    {
                        organisationsWithNoActiveScopeForEveryYear.Add(organisation);
                    }```